### PR TITLE
Updated login to include failure flash message

### PR DESCRIPTION
**Pull request by jeremyharris**

### CODE REVIEW

File: docs/Migration-from-the-AuthComponent.md [hunk 1]
```diff
@@ -33,9 +33,17 @@ The AuthenticationMiddleware will handle checking and setting the identity based
 ```php
 function login()
 {
+    $result = $this->Authentication->getResult();
+
     // regardless of POST or GET, redirect if user is logged in
-    if ($this->Authentication->getResult()->isValid()) {
-        return $this->redirect('/');
+    if ($result->isValid()) {
+        $redirect = $this->request->getQuery('redirect', ['controller' => 'Pages', 'action' => 'display', 'home']);
```
Comment (ADmad) [inline docs/Migration-from-the-AuthComponent.md]: Perhaps use string URL instead of array? We are trying to keep this plugin decoupled from Cake.

Comment (ADmad) [reply docs/Migration-from-the-AuthComponent.md]: But meh, the whole example is Cake specific.

Comment (dereuromark) [reply docs/Migration-from-the-AuthComponent.md]: I would rather see more URL arrays then just strings, especially in the CakePHP scope.

Comment (markstory) [reply docs/Migration-from-the-AuthComponent.md]: Or named routes :bike:

**Comment by markstory** [comment]
Merging as the example code is a useful improvement over what as there before.

**Pull request merged**